### ISSUE TITLE
Executing scripts in parallel - problem with jobs with multiple steps

### ISSUE THREAD

**Issue by iperic95** [olahallengren/sql-server-maintenance-solution#434]
Hi,

I have some problems with executing OLA maintenance soultion with parallel. I created 2 jobs, each having steps : Integrity Check, Rebuild indexes and Differential backup using OLA stored procedures with parametar 'Parallel' When executing it, the result is twice generated differential backup for all databases. The scenario goes in way, first job runs all steps and creates differential backup for all databases on last step, while the other job at same time is still running integrity check for big database and then when it finishes, runs rebuild and backup with  generatating one more backup for same databases, which have been already done by other job. I have 3 databases ( two small and one big database).  I am not fully understanding why second job just do not skip other two steps ( rebuild indexes and differential backup) as it has already been done by other job (session), than it backuping it all over again.

If you are wondering why I am not going for diving each step in job, that's because I want to ensure that no backups are made if integrity check  step fails. So with this configuration, first step Integrity check will not go on next step if it fails.

I don't know if it is even intended to run paralleling like these ( multiple steps in job), so I am wondering if I am doing domething wrong or is there some other solution to overcome this problem?

**Comment by olahallengren** [comment]
Thank you for describing your scenario.

Unfortunately it does not work with the parallel backup feature.

The parallel backup will only work if the backup steps for the different jobs run at the same time. If one job execution has completed, and then another one starts, it is considered a new backup.
